v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/utils/KafkaConfig.java

```diff
@@ -87,10 +87,9 @@ private static Map<String, Object> from(
           SslConfigs.SSL_KEY_PASSWORD_CONFIG,
           FileAwareFactoryFn.SECRET_MANAGER_VALUE_PREFIX + keyPasswordSecretId);
       properties.put(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "");
-    } else if (authMode.equals(KafkaAuthenticationMethod.SASL_MECHANISM)) {
-      properties.put(SaslConfigs.SASL_MECHANISM, KafkaAuthenticationMethod.SASL_MECHANISM);
-      properties.put(
-          CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, KafkaAuthenticationMethod.SASL_PLAIN);
+    } else if (authMode.equals(KafkaAuthenticationMethod.PLAIN)) {
+      properties.put(SaslConfigs.SASL_MECHANISM, KafkaAuthenticationMethod.PLAIN);
```

Comment:
```suggestion
      properties.put(SaslConfigs.SASL_MECHANISM, "PLAIN");
```
I suggest just hard-coding "PLAIN" here, or introducing another constant. What's set as `sasl.mechanism` property value in Kafka config doesn't have to match the enum value we use in the template's Authentication Mode parameter (and as discussed offline, it won't: we'll use `SASL_PLAIN` as the enum value, but here we have to set the property to `PLAIN`).